src/loaders/google_map_loader.js

```diff
@@ -1,87 +1,89 @@
-      reject(new Error('google map cannot be loaded outside browser env'));
-      return;
-    }
+const loadScript = url => {
+  const script = document.createElement('script');
 
-    if (window.google && window.google.maps) {
-      resolve(window.google.maps);
-      return;
-    }
+  script.type = 'text/javascript';
+  script.async = true;
+  script.src = url;
+  document.querySelector('head').appendChild(script);
 
-    if (typeof window._$_google_map_initialize_$_ !== 'undefined') {
-      reject(new Error('google map initialization error'));
+  return new Promise(resolve => {
+    if (currentResolver) {
+      window.removeEventListener(EVENT_GMAPS_LOADED, currentResolver);
     }
-
-    window._$_google_map_initialize_$_ = () => {
-      delete window._$_google_map_initialize_$_;
-      resolve(window.google.maps);
+    currentResolver = () => {
+      resolve();
     };
+    window.addEventListener(EVENT_GMAPS_LOADED, currentResolver);
+  });
+};
 
-    if (process.env.NODE_ENV !== 'production') {
-      if (Object.keys(bootstrapURLKeys).indexOf('callback') > -1) {
-        const message = `"callback" key in bootstrapURLKeys is not allowed,
-                          use onGoogleApiLoaded property instead`;
-        // eslint-disable-next-line no-console
-        console.error(message);
-        throw new Error(message);
-      }
-    }
-
-    const params = Object.keys(bootstrapURLKeys).reduce(
-      (r, key) => `${r}&${key}=${bootstrapURLKeys[key]}`,
-      ''
-    );
+const loadGoogleMaps = bootstrapURLKeys =>
+  new Promise(async resolve => {
+    lastScriptUrl = getScriptUrl(bootstrapURLKeys);
+    await loadScript(lastScriptUrl);
+    resolve(window.google.maps);
+  });
 
-    const baseUrl = getUrl(bootstrapURLKeys.region);
-    const libraries = heatmapLibrary ? '&libraries=visualization' : '';
+export default bootstrapURLKeys => {
+  if (typeof window === 'undefined') {
+    throw new Error('google map cannot be loaded outside browser env');
+  }
 
-    $script_(
-      `${baseUrl}${API_PATH}${params}${libraries}`,
-      () =>
-        typeof window.google === 'undefined' &&
-        reject(new Error('google map initialization error (not loaded)'))
-    );
-  });
+  if (process.env.NODE_ENV !== 'production') {
+    if (Object.keys(bootstrapURLKeys).includes('callback')) {
+      const message = `'callback' key in bootstrapURLKeys is not allowed, use onGoogleapiLoadedPromise property instead`;
+      // eslint-disable-next-line no-console
+      console.error(message);
+      throw new Error(message);
+    }
+  }
+  if (googleMapsPromise) {
+    if (lastScriptUrl !== getScriptUrl(bootstrapURLKeys)) {
+      destroyOldGoogleMapsInstance(lastBaseUrl);
+      googleMapsPromise = loadGoogleMaps(bootstrapURLKeys);
+    }
+    return googleMapsPromise;
+  }
 
-  resolveCustomPromise_(loadPromise_);
+  googleMapsPromise = loadGoogleMaps(bootstrapURLKeys);
+  lastBaseUrl = getBaseUrl(bootstrapURLKeys.region);
```

Comment:
Are you using lastBaseUrl after you define it? I don't see that 🤔

Reply:
Yes, i am using that variable for finding and destructing loaded scripts from Google maps from previous instance. It's because of returned different domains from `getBaseUrl` function.